src/main/java/pedido/Pedido.java

```diff
@@ -1,51 +1,113 @@
+                    totalItem += precoIngrediente.getValue();
+                }
+            }
+            totalItem = totalItem * itemPedido.getShake().getTipoTamanho().multiplicador;
+            for (Map.Entry<Ingrediente, Double> precoIngrediente : cardapio.getPrecos().entrySet()) {
+                for (Adicional adicional : itemPedido.getShake().getAdicionais()) {
+                    if (adicional.equals(precoIngrediente.getKey())) {
+                        totalItem += precoIngrediente.getValue();
+                    }
+                }
+            }
+            totalItem = totalItem * itemPedido.getQuantidade();
+            total += totalItem;
+        }
         return total;
     }
 
-    public void adicionarItemPedido(ItemPedido itemPedidoAdicionado){
-        //TODO
+    public void adicionarItemPedido(ItemPedido itemPedidoAdicionado) {
+        sortAdicionais(itemPedidoAdicionado);
+        boolean added = false;
+        for (ItemPedido item : itens) {
+            if (!added) {
+                if (item.getShake().equals(itemPedidoAdicionado.getShake())) {
+                    item.setQuantidade(item.getQuantidade() + itemPedidoAdicionado.getQuantidade());
+                    added = true;
+                }
+            }
+        }
+
+        if (!added) {
+            itens.add(itemPedidoAdicionado);
+        }
     }
 
     public boolean removeItemPedido(ItemPedido itemPedidoRemovido) {
-        //substitua o true por uma condição
-        if (true) {
-            //TODO
-        } else {
-            throw new IllegalArgumentException("Item nao existe no pedido.");
+        sortAdicionais(itemPedidoRemovido);
+        ItemPedido itemToRemove = null;
+        boolean removed = false;
+        for (ItemPedido item : itens) {
+            if (!removed) {
+                if (item.getShake().equals(itemPedidoRemovido.getShake())) {
+                    if (item.getQuantidade() == 1) {
+                        itemToRemove = item;
+                        removed = true;
+                    } else {
+                        item.setQuantidade(item.getQuantidade() - 1);
+                        removed = true;
+                    }
+                }
+            }
         }
+        if (itemToRemove != null)
+            itens.remove(itemToRemove);
+        if (!removed)
+            throw new IllegalArgumentException("Item nao existe no pedido.");
         return false;
     }
 
+    public void sortAdicionais(ItemPedido pedido) {
+        if (pedido.getShake().getAdicionais() != null) {
+            if (pedido.getShake().getAdicionais().size() > 0) {
+                pedido.getShake().getAdicionais().sort(new Comparator<Ingrediente>() {
+                    @Override
+                    public int compare(Ingrediente o1, Ingrediente o2) {
+                        return o1.obterTipo().toString().compareTo(o2.obterTipo().toString());
+                    }
+                });
+            }
+        }
+    }
+
```

Comment:
Talvez fique melhor você order os adicionais direto no construtor. Porque dai você já garante que os adicionais estarão ordenados e não precisa chamar esse método todas as vezes que for manipular os adicionais.